marlowe-playground-client/src/Contrib/Data/Array/Builder.purs

```diff
@@ -0,0 +1,40 @@
+module Contrib.Data.Array.Builder where
+
+import Prelude
+
+-- | Monoidal array builder which should
+-- | have much better performance than concatenation.
+-- |
+-- | Use it for relatively small arrays (length < 10000).
+-- | It is not stack safe.
+newtype Builder a
+  = Builder (Array a → Array a)
+
+instance semigroupBuilder ∷ Semigroup (Builder a) where
+  append (Builder b1) (Builder b2) = Builder (b1 <<< b2)
+
+instance monoidBuilder ∷ Monoid (Builder a) where
+  mempty = Builder identity
+
+foreign import unsafeAppend ∷ ∀ a. Array a → Array a → Array a
```

Comment:
Just an FYI that we will be enforcing ascii syntax with the switch to `purs-tidy` (although I could be convinced to make it do unicode instead - it is more concise for sure)

Reply:
This is an accident. I thought that purty is going to drop this. This is going to be removed by `purs-tidy` for sure. Should I fix it before merge?